src/v/cluster/rm_stm.h

```diff
@@ -371,7 +371,9 @@ class rm_stm final : public persisted_stm {
         absl::flat_hash_map<model::producer_identity, seq_entry> seq_table;
     };
 
-    struct mem_state {
+    struct mem_state
```

Comment:
let's use data transfer object to transfer the state to the followers instead of serializing mem_state. It creates a coupling between in memory state and on disk state and makes it very hard to evolve the format independently